experiments/Schema/src/Query.jl

```diff
@@ -22,156 +22,246 @@ module QueryLib
   struct Query
     dom::Types
     codom::Types
-    dom_names::Array{String,1}
-    codom_names::Array{String,1}
-    query::String
-    type::String
+    dom_names::Array{Int,1}
+    codom_names::Array{Int,1}
+    tables::Array{String,1}
+    fields::Array{Tuple{Int,String},1}
+    connections::Array{Tuple{Int,Int},1}
```

Comment:
Should we use a `LightGraphs.SimpleDigraph` here? It won't add a dependency since Catlab already requires LightGraphs for WDs.

Reply:
Also the docstring of this struct should describe what the fields mean. The only one that is not obvious is the connections, there is an edge `e[i,j]` in this digraph iff `P(i,g)`?

Reply:
Yeah, LightGraphs would capture the intuition much better! And I'm not sure what `P(i,j)` is here, but there exists an edge `e[i,j]` iff `i` and `j` are connected in the wiring diagram. The algorithm basically re-routes connections to `mcopy`, `mmerge`, and `id` while preserving the relationships they impose so that all nodes in the graph have an associated table.

Reply:
After taking a further look at it, we can add a SimpleDigraph to just replace the `connections` field, but we could also replace `tables` and `fields` if we had a julia graph library that supports ports. I haven't found a library that supports that yet, but it'd probably make the representation more intuitive.